### ISSUE TITLE
Payment fails with Drop-in UI using a Vaulted card & 3D Secure enabled

### ISSUE THREAD

**Issue by Nonepse** [braintree/braintree-ios-drop-in#114]
### General information

* SDK/Library version: 6.4.0
* Environment: Production
* iOS Version and Device: N/A
* Integration type and version: N/A

### Issue description

I've been pulling my hairs for some time, because a payment using a vaulted card with 3DS enabled fails.
Looking at the Android & Web flow I noticed the 3DS flow was triggered when using a vaulted card, but on iOS the handler block was executed without any 3DS flow.

I won't give all the details of the error because it's actually a known issue https://github.com/braintree/braintree-ios-drop-in/issues/93 (also https://github.com/braintree/braintree-ios-drop-in/issues/26).
Shame on me for not reading the closed issues.

Since fixing it has been a known issue for some time now, I am assuming it's not straightforward.

**I am opening this issue to ask for that difference between the SDKs to be documented.**

The solution I used was to trigger the 3DS flow if the `threeDSecureInfo.wasVerified` property on the payment nonce is `false`.
Here is my code in Swift, inspired from the code in the [demo project in Objective-C](https://github.com/braintree/braintree-ios-drop-in/blob/master/DropInDemo/Features/Drop%20In/BraintreeDemoDropInViewController.m#L225):

Since the requirements for that to work are not documented, can someone confirm this flow is valid?

```swift
private lazy var braintreeRequest: BTDropInRequest = {
    let request = BTDropInRequest()

    request.vaultManager = true
    request.threeDSecureVerification = true
    request.amount = self.viewModel.amount

    return request
}()

private func showBraintreeDropIn() {
    guard let clientToken = self.viewModel.clientToken else {
        assertionFailure("It's a developer error to trigger Braintree UI if the token is not yet fetched.")
        return
    }

    BTUIKAppearance.sharedInstance().postalCodeFormFieldKeyboardType = .default

    let dropIn = BTDropInController(
        authorization: clientToken,
        request: self.braintreeRequest
    ) { controller, result, error in

        if let result = result, let paymentMethodNonce = result.paymentMethod {

            if self.nonceRequiresThreeDSecureVerification(nonce: paymentMethodNonce) {
                self.performThreeDSecureVerification(nonce: paymentMethodNonce, token: clientToken)
            } else {
                self.viewModel.sendNonce(paymentMethodNonce.nonce)
            }

        } else {
            // Your error management
        }

        controller.dismiss(animated: true, completion: nil)
    }

    self.present(dropIn!, animated: true, completion: nil)
}

private final func nonceRequiresThreeDSecureVerification(nonce: BTPaymentMethodNonce) -> Bool {
    if let cardNonce = nonce as? BTCardNonce {
        return !cardNonce.threeDSecureInfo.wasVerified
    }

    return false
}

private final func performThreeDSecureVerification(nonce: BTPaymentMethodNonce, token: String) {
    guard let apiClient = BTAPIClient(authorization: token) else {
        // Your error management
        return
    }

    let paymentFlowDriver = BTPaymentFlowDriver(apiClient: apiClient)
    paymentFlowDriver.viewControllerPresentingDelegate = self

    let request = BTThreeDSecureRequest()
    request.amount = NSDecimalNumber(string: self.viewModel.amount)
    request.nonce = nonce.nonce

    paymentFlowDriver.startPaymentFlow(request) { [weak self] result, error in
        guard let `self` = self else { return }

        if let result = result as? BTThreeDSecureResult {
            self.viewModel.sendNonce(result.tokenizedCard.nonce)
        }
    }
}

// MARK: BTViewControllerPresentingDelegate

func paymentDriver(_ driver: Any, requestsPresentationOf viewController: UIViewController) {
    self.present(viewController, animated: true)
}

func paymentDriver(_ driver: Any, requestsDismissalOf viewController: UIViewController) {
    self.dismiss(animated: true)
}
```

Also hope this can help other devs coming after me ;)

**Comment by crookedneighbor** [comment]
Thanks for the report. This is on our roadmap to align the sdks to all have the same behavior.

There is a workaround mentioned here to run the 3ds result yourself: https://github.com/braintree/braintree-ios-drop-in/issues/93

**Comment by sestevens** [comment]
This is resolved in version 7.3.0.
